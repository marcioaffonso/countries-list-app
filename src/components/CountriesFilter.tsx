import React from 'react';
import { View, Text, StyleSheet, TextInput } from 'react-native';
import PropTypes from 'prop-types';

export default function CountriesFilter({ onFilterChange }) {
  return (
    <View style={styles.filterContainer}>
      <Text style={styles.filterText}>Filter by name:</Text>
      <TextInput
        style={styles.filterInput}
        onChangeText={onFilterChange}
      />
    </View>
  );
};

CountriesFilter.propTypes = {
  onFilterChange: PropTypes.func,
};

const styles = StyleSheet.create({
  filterContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal: 10,
    marginBottom: 10,
    minHeight: 50,
    maxHeight: 50,
    height: 50,
  },
  filterText: {
    flex: 1,
    fontSize: 20
  },
  filterInput: {
    flex: 1,
    borderRadius: 10,
    borderWidth: 1,
    backgroundColor: '#fff',
    borderColor: '#000',
    fontSize: 20,
    paddingVertical: 2,
    paddingHorizontal: 5
  },
});
