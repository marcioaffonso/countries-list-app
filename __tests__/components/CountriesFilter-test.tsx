import React from 'react';
import { TextInput } from 'react-native';
import renderer from 'react-test-renderer';
import { jest, expect, test } from '@jest/globals';
import CountriesFilter from '../../src/components/CountriesFilter';

const onFilterhMock = jest.fn();

test('renders correctly', () => {
  const tree = renderer.create(<CountriesFilter onFilterChange={onFilterhMock} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders App loading correctly', () => {
  const mockValue = 'the-value';
  const tree = renderer.create(<CountriesFilter onFilterChange={onFilterhMock} />);
  tree.root.findByType(TextInput).props.onChangeText(mockValue);
  expect(onFilterhMock).toBeCalledWith(mockValue);
});
