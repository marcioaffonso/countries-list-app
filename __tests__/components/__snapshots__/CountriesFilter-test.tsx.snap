// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginHorizontal": 10,
      "maxHeight": 50,
      "minHeight": 50,
    }
  }
>
  <Text
    style={
      {
        "flex": 1,
        "fontSize": 20,
      }
    }
  >
    Filter by name:
  </Text>
  <TextInput
    onChangeText={[MockFunction]}
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderRadius": 10,
        "borderWidth": 1,
        "flex": 1,
        "fontSize": 20,
        "paddingHorizontal": 5,
        "paddingVertical": 2,
      }
    }
  />
</View>
`;
